src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -637,11 +687,13 @@ ucs_status_t uct_dc_mlx5_ep_flush(uct_ep_h tl_ep, unsigned flags,
         }
 
         if (ep->flags & UCT_DC_MLX5_EP_FLAG_FLUSH_CANCEL) {
+            uct_dc_mlx5_iface_dci_put(iface, ep->dci);
```

Comment:
why do we need to put here? `UCT_DC_MLX5_CHECK_RES` on line 683 looks strange, we did all the checks for ep DCI in the beginning of this func